src/runtime_action.rs

```diff
@@ -155,13 +155,25 @@ impl RuntimeAction {
         match self {
             RuntimeAction::RateLimit(rl_action) => {
                 let descriptor = rl_action.build_descriptor()?;
+                let known_attrs = rl_action.get_known_attributes()?;
+
                 if descriptor.entries.is_empty() {
                     debug!("build_message(rl): empty descriptors");
                     Ok(None)
                 } else {
+                    let domain = known_attrs
+                        .get("ratelimit.domain")
+                        .cloned()
```

Comment:
You should be able to "merge these 2 calls" into a single [`.remove`](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.remove) and avoid the `clone()`ing of the string. Not a huge deal at all, but so you know :)